lib/crapto/xor.nit

```diff
@@ -0,0 +1,150 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Cryptographic attacks and utilities for XOR-based algorithms.
+module xor
+
+import combinations
+import crypto
+import english_utils
+
+redef class SingleByteXorCipher
+	# Tries to find key using frequency analysis on all possible plaintexts.
+	# Populates `self.key`
+	fun find_key do
+
+		# Accumulate best result
+		var max = 0.0
+		var best = 0.to_b
+
+		# Iterate on possible values for a byte
+		for b in [0 .. 255] do
+			# Need `Bytes` to pass to xor
+			var xor_b = new Bytes.with_capacity(1)
```

Comment:
Nitpick: I'd suggest you allocate `xor_b` outside of the loop and reset it at the beginning of each loop for a bit more performance.